hat/assets/js/apps/Iaso/components/forms/InputComponent.tsx

```diff
@@ -0,0 +1,255 @@
+import React, { ReactNode, useState } from 'react';
+import { Box } from '@material-ui/core';
+import {
+    TextInput,
+    PasswordInput,
+    NumberInput,
+    Radio,
+    Checkbox,
+    ArrayFieldInput,
+    SearchInput,
+    // @ts-ignore
+    translateOptions,
+    Select,
+    useSafeIntl,
+    IntlMessage,
+} from 'bluesquare-components';
+import MESSAGES from '../../domains/forms/messages';
+import { DropdownOptions } from '../../types/utils';
+
+type Option = DropdownOptions<string | number>;
+
+export type InputComponentType =
+    | 'text'
+    | 'email'
+    | 'password'
+    | 'number'
+    | 'radio'
+    | 'checkbox'
+    | 'arrayInput'
+    | 'search'
+    | 'select';
+
+export type InputComponentProps = {
+    type: InputComponentType;
+    keyValue: string;
+    value?: any;
+    errors?: string[];
+    // eslint-disable-next-line no-unused-vars
+    onChange?: (key: string, value: any) => void;
+    options?: any[];
+    disabled?: boolean;
+    multiline?: boolean;
+    clearable?: boolean;
+    label?: IntlMessage;
+    labelString?: string;
+    required?: boolean;
+    onEnterPressed?: () => void;
+    withMarginTop?: boolean;
+    multi?: boolean;
+    uid?: string;
+    loading?: boolean;
+    // eslint-disable-next-line no-unused-vars
+    getOptionLabel?: (option: Option) => string;
+    getOptionSelected?: (
+        // eslint-disable-next-line no-unused-vars
+        option: Option,
+        // eslint-disable-next-line no-unused-vars
+        value: Option,
+    ) => boolean;
+    renderOption?: (
+        // eslint-disable-next-line no-unused-vars
+        option: Option,
+        // eslint-disable-next-line no-unused-vars
+        { inputValue }: { inputValue: string },
+    ) => ReactNode;
+    className?: string;
+    helperText?: string;
+    min?: number;
+    max?: number;
+    blockForbiddenChars?: boolean;
+    onErrorChange?: () => void;
+    numberInputOptions?: {
+        min?: number;
+        max?: number;
+        decimalScale?: number;
+    };
+};
+
+const InputComponent: React.FC<InputComponentProps> = ({
+    type = 'text',
+    keyValue,
+    value,
+    errors = [],
+    onChange = () => null,
+    options = [],
+    disabled = false,
+    multiline = false,
+    clearable = true,
+    label,
+    labelString = null,
+    required = false,
+    onEnterPressed = () => null,
+    withMarginTop = true,
+    multi = false,
+    uid,
+    loading = false,
+    getOptionLabel,
+    getOptionSelected,
+    renderOption,
+    className = '',
+    helperText,
+    min,
+    max,
+    blockForbiddenChars = false,
+    onErrorChange = () => null,
+    numberInputOptions,
+}) => {
+    const [displayPassword, setDisplayPassword] = useState(false);
+    const { formatMessage } = useSafeIntl();
+
+    const toggleDisplayPassword = () => {
+        setDisplayPassword(!displayPassword);
+    };
```

Comment:
smart ass version (this way you don't have to worry about `useCallback`):

```suggestion
    const toggleDisplayPassword = () => {
        setDisplayPassword(value=>!value);
    };
```